apollo-compiler/src/main/kotlin/com/apollographql/android/compiler/ir/Fragment.kt

```diff
@@ -10,16 +10,23 @@ import javax.lang.model.element.Modifier
 data class Fragment(
     val fragmentName: String,
     val source: String,
-    val typeCondition:String,
+    val typeCondition: String,
     val possibleTypes: List<String>,
     val fields: List<Field>,
     val fragmentSpreads: List<String>,
     val inlineFragments: List<InlineFragment>,
     val fragmentsReferenced: List<String>
 ) : CodeGenerator {
+
   /** Returns the Java interface that represents this Fragment object. */
   override fun toTypeSpec(context: CodeGenerationContext): TypeSpec =
-      SchemaTypeSpecBuilder(formatClassName(), fields, fragmentSpreads, inlineFragments, context)
+      SchemaTypeSpecBuilder(
+          typeName = formatClassName(),
+          fields = listOf(Field("__typename", "__typename", "String!")) + fields,
```

Comment:
nvm, I guess that is just to the query. But this to actually add it as a field. 👍